pipelines/library/src/ParseVersion.groovy

```diff
@@ -0,0 +1,52 @@
+import common.VersionInfo
+import groovy.json.JsonOutput
+
+import java.util.regex.Matcher
+
+class ParseVersion {
+
+    static void main(String[] args) {
+        OptionAccessor parsedArgs = parseArgs(args)
+        VersionInfo version
+        if (parsedArgs.s) {
+            version = parseFromStdIn(args)
+        } else {
+            version = new VersionInfo().parse(args[args.length - 2], args[args.length - 1])
+        }
+
+        printVersion(parsedArgs, version)
+    }
+
+    private static void printVersion(OptionAccessor parsedArgs, VersionInfo version) {
+        if (parsedArgs.f) {
+            def toPrint = ((String) parsedArgs.getProperty("f")).split(",")
+            toPrint.each { arg ->
+                println(version.getProperty(arg))
+            }
+        } else {
+            println(JsonOutput.prettyPrint(JsonOutput.toJson(version)))
+        }
+    }
+
+    private static VersionInfo parseFromStdIn(String[] args) {
+        def reader = System.in.newReader()
+        def line
+        while ((line = reader.readLine()) != null) {
+            Matcher matcher = (line =~ /.*\(build (?<version>.*)\).*/)
+            if (matcher.matches()) {
+                VersionInfo version = new VersionInfo().parse(matcher.group("version"), args[args.length - 1])
+                return version
+            }
+        }
+        return null
+    }
+
+    private static OptionAccessor parseArgs(String[] args) {
+        CliBuilder cliBuilder = new CliBuilder()
+
+        cliBuilder.s('read input from stdin', args: 0)
```

Comment:
`s` and `f`?

Reply:
same as before, this defines you can give arguments '-s' and '-f'